app/src/main/java/org/mifos/selfserviceapp/login/LoginPresenter.java

```diff
@@ -0,0 +1,52 @@
+package org.mifos.selfserviceapp.login;
+
+import android.util.Log;
+
+import org.mifos.selfserviceapp.data.User;
+import org.mifos.selfserviceapp.api.DataManager;
+import org.mifos.selfserviceapp.presenters.BasePresenter;
+import org.mifos.selfserviceapp.utils.PrefManager;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * @author Vishwajeet
+ * @since 05/06/16
+ */
+
+public class LoginPresenter extends BasePresenter<LoginView> {
+
+    DataManager mDataManager;
+
+    protected LoginPresenter(DataManager dataManager) {
+       mDataManager = dataManager;
+    }
+
+
+    public void login(String username, String password) {
+        Call<User> call = mDataManager.login(username, password);
+        getMvpView().showProgress();
+        call.enqueue(new Callback<User>() {
+            @Override
+            public void onResponse(Response<User> response) {
+                getMvpView().hideProgress();
+                getMvpView().onLoginSuccessful(response);
```

Comment:
Handle the response in the presenter and then delegate a call to the view about what needs to be done.